src/cli.rs

```diff
@@ -1,50 +1,54 @@
-                          .arg(Arg::with_name("v")
-                               .short("v")
-                               .multiple(true)
-                               .help("Sets the level of verbosity"))
-                          .get_matches();
-
-    // Gets a value for config if supplied by user, or defaults to "default.conf"
-    //let config = matches.value_of("config").unwrap_or("default.conf");
-    //println!("Value for config: {}", config);
-
-    // Calling .unwrap() is safe here because "INPUT" is required (if "INPUT" wasn't
-    // required we could have used an 'if let' to conditionally get the value)
-    //println!("Using input file: {}", matches.value_of("INPUT").unwrap());
-
-    // Vary the output based on how many times the user used the "verbose" flag
-    // (i.e. 'myprog -v -v -v' or 'myprog -vvv' vs 'myprog -v'
-    match matches.occurrences_of("v") {
-        0 => println!("No verbose info"),
-        1 => println!("Some verbose info"),
-        2 => println!("Tons of verbose info"),
-        3 | _ => println!("Don't be crazy"),
-    }
+                    .version("0.1.0")
+                    .author("Marek C. <countsmarek@gmail.com>")
+                    .about("Does awesome things! very awesome.")
+                    .arg(Arg::with_name("config")
+                        .short("c")
+                        .long("config")
+                        .value_name("FILE")
+                        .help("Sets a custom config file")
+                        .takes_value(true))
+                    .arg(Arg::with_name("v")
+                        .short("v")
+                        .multiple(true)
+                        .help("Sets the level of verbosity"))
+                    .arg(Arg::with_name("daemon")
+                        .short("d")
+                        .long("daemon")
+                        .multiple(false)
+                        .help("Setting this flag will enable running as a daemon"))
+                    .get_matches();
+
+    if matches.is_present("daemon") {
 
-    // You can handle information about subcommands by requesting their matches by name
-    // (as below), requesting just the name used, or both at the same time
-    if let Some(matches) = matches.subcommand_matches("test") {
-        if matches.is_present("debug") {
-            println!("Printing debug info...");
-        } else {
-            println!("Printing normally...");
+        println!("Running as daemon");
+        let daemonize = Daemonize::new().pid_file("/var/run/gary.pid")
+                                        .chown_pid_file(true)
+                                        .working_directory("/var/run/")
+                                        .user("root")
+                                        .group("daemon") // Group name
+                                        .group(2) // Or group id
+                                        .privileged_action(|| "Executed before drop privileges");
```

Comment:
This also doesn't need to be here (aside from the `);`).  This would be used to open a low port or start up a daemon that does limited stuff before switching user to non-root.  We're staying as root (for now), so leaving it as a noop like it is (right?) or removing it are equally fine.